src/pipecat/transports/base_output.py

```diff
@@ -46,80 +47,71 @@ def __init__(self, params: TransportParams, **kwargs):
 
         self._params = params
 
-        # Task to process incoming frames so we don't block upstream elements.
-        self._sink_task = None
-
-        # Task to process incoming frames using a clock.
-        self._sink_clock_task = None
-
-        # Task to write/send audio and image frames.
-        self._video_out_task = None
-
-        # These are the images that we should send at our desired framerate.
-        self._video_images = None
-
-        # Output sample rate. It will be initialized on StartFrame.
-        self._sample_rate = 0
-        self._resampler = create_default_resampler()
-
-        # Chunk size that will be written. It will be computed on StartFrame
-        self._audio_chunk_size = 0
-        self._audio_buffer = bytearray()
-
-        self._stopped_event = asyncio.Event()
-
-        # Indicates if the bot is currently speaking.
-        self._bot_speaking = False
+        # We will have one media sender per output frame destination. This allow
+        # us to send multiple streams at the same time if the transport allows
+        # it.
+        self._media_senders: Dict[Any, "BaseOutputTransport.MediaSender"] = {}
 
     @property
     def sample_rate(self) -> int:
-        return self._sample_rate
+        sender = self._media_senders.get(None, None)
+        return sender.sample_rate if sender else 0
+
+    @property
+    def audio_chunk_size(self) -> int:
+        sender = self._media_senders.get(None, None)
+        return sender.audio_chunk_size if sender else 0
```

Comment:
Shouldn't we keep `_sample_rate` and `_audio_chunk_size` as attributes of the class ? It feels a bit odd to retrieve these values from the default media sender.

Reply:
I did it like this in case each media sender had a different sample rate. It's true that we could compute these values in the base class and pass them to the media senders instead as currently they all compute the same.

Done!